### PR TITLE
Concurrent scheduler refactor

### PR DESCRIPTION
**Pull request by thomastanck**
Move scheduler to its own file in anticipation for more schedulers to be (possibly) added.

### CODE REVIEW

**Comment by thomastanck** [comment]
Test fails because of spurious test: #688.

File: src/vm/svml-machine.ts [hunk 1]
```diff
@@ -584,8 +585,59 @@ function INTERNAL_FUNCTION_CALL() {
...
+}
+
+// Schedule current thread for later execution
+// You will want to pop another thread, restore thread state, etc.
+// after calling this, as the current thread state should not be running
+function PAUSE_THREAD() {
+  // Save state to threads map
+  threads.set(currentThreadId, [OS, ENV, PC, P, RTS, TOP_RTS])
+
+  // Pause thread in scheduler, and get new one
```
Comment (jonchan51) [inline src/vm/svml-machine.ts:613]:
```suggestion
  // Pause thread in scheduler
```

File: src/vm/svml-machine.ts [hunk 2]
```diff
@@ -584,8 +585,59 @@ function INTERNAL_FUNCTION_CALL() {
   }
 }
 
-function SET_TO() {
-  TO = Math.ceil((0.5 + Math.random() * 0.5) * TO_MAX)
+let scheduler: Scheduler = new RoundRobinScheduler()
+const threads: Map<ThreadId, any> = new Map()
```
Comment (jonchan51) [inline src/vm/svml-machine.ts:589]: hm is it possible to define a tuple type to represent a thread? or perhaps just a object

**Comment by thomastanck** [comment]
Changes applied

**Comment by zhengqunkoo** [comment]
LGTM

**Pull request merged**